flow/core/kernel/vehicle/traci.py

```diff
@@ -949,18 +950,22 @@ def _prev_edge_followers(self, veh_id, edge_dict, lane, num_edges):
 
         return tailway, follower
 
-    def apply_acceleration(self, veh_ids, acc):
+    def apply_acceleration(self, veh_ids, acc, smooth=False):
         """See parent class."""
-        # to hand the case of a single vehicle
+        # to handle the case of a single vehicle
         if type(veh_ids) == str:
             veh_ids = [veh_ids]
             acc = [acc]
 
         for i, vid in enumerate(veh_ids):
             if acc[i] is not None and vid in self.get_ids():
+                self.__vehicles[vid]["accel"] = acc[i]
                 this_vel = self.get_speed(vid)
                 next_vel = max([this_vel + acc[i] * self.sim_step, 0])
-                self.kernel_api.vehicle.slowDown(vid, next_vel, 1e-3)
+                if smooth:
+                    self.kernel_api.vehicle.slowDown(vid, next_vel, 1e-3)
```

Comment:
Is keeping the `smooth` option necessary? I'm not sure a 1e-3s slowdown changes anything when we are dealing with .1 or .5s timesteps

Reply:
@eugenevinitsky do you have any thoughts on this? should we default master to using `setSpeed()`?

Reply:
I think @nathanlct is right here. Let's keep things as is for now, and if we want to change it later we can do it.